libcrux_crypto/src/crypto.rs

```diff
@@ -328,247 +324,66 @@ impl OpenMlsCrypto for CryptoProvider {
-        HashType::Sha2_512 => libcrux::hkdf::Algorithm::Sha512,
-    }
-}
-
-fn aead_key(alg: AeadType, key: &[u8]) -> Result<libcrux::aead::Key, CryptoError> {
-    let key = match alg {
-        AeadType::Aes128Gcm => {
-            const ALG: libcrux::aead::Algorithm = libcrux::aead::Algorithm::Aes128Gcm;
-            let key: [u8; ALG.key_size()] =
-                key.try_into().map_err(|_| CryptoError::InvalidLength)?;
-            libcrux::aead::Key::Aes128(libcrux::aead::Aes128Key(key))
-        }
-        AeadType::Aes256Gcm => {
-            const ALG: libcrux::aead::Algorithm = libcrux::aead::Algorithm::Aes256Gcm;
-            let key: [u8; ALG.key_size()] =
-                key.try_into().map_err(|_| CryptoError::InvalidLength)?;
-            libcrux::aead::Key::Aes256(libcrux::aead::Aes256Key(key))
-        }
-        AeadType::ChaCha20Poly1305 => {
-            const ALG: libcrux::aead::Algorithm = libcrux::aead::Algorithm::Chacha20Poly1305;
-            let key: [u8; ALG.key_size()] =
-                key.try_into().map_err(|_| CryptoError::InvalidLength)?;
-            libcrux::aead::Key::Chacha20Poly1305(libcrux::aead::Chacha20Key(key))
-        }
-    };
-
-    Ok(key)
-}
-
-fn sig(alg: SignatureScheme, sig: &[u8]) -> Result<libcrux::signature::Signature, CryptoError> {
-    match alg {
-        SignatureScheme::ECDSA_SECP256R1_SHA256 => {
-            let decoded: [u8; 64] = der_decode(sig)?
-                .try_into()
-                .map_err(|_| CryptoError::InvalidSignature)?;
-
-            Ok(libcrux::signature::Signature::EcDsaP256(
-                libcrux::signature::EcDsaP256Signature::from_bytes(
-                    decoded,
-                    libcrux::signature::Algorithm::EcDsaP256(
-                        libcrux::signature::DigestAlgorithm::Sha256,
-                    ),
-                ),
-            ))
-        }
-        SignatureScheme::ED25519 => libcrux::signature::Ed25519Signature::from_slice(sig)
-            .map(libcrux::signature::Signature::Ed25519)
-            .map_err(|e| match e {
-                libcrux::signature::Error::InvalidSignature => CryptoError::InvalidSignature,
-                _ => CryptoError::CryptoLibraryError,
-            }),
-        _ => Err(CryptoError::UnsupportedSignatureScheme),
-    }
-}
+fn hpke_config(config: HpkeConfig) -> hpke_rs::Hpke<HpkeLibcrux> {
+    let kem = hpke_kem(config.0);
+    let kdf = hpke_kdf(config.1);
 
-fn sig_alg(alg: SignatureScheme) -> Result<libcrux::signature::Algorithm, CryptoError> {
-    match alg {
-        SignatureScheme::ECDSA_SECP256R1_SHA256 => Ok(libcrux::signature::Algorithm::EcDsaP256(
-            libcrux::signature::DigestAlgorithm::Sha256,
-        )),
-        SignatureScheme::ED25519 => Ok(libcrux::signature::Algorithm::Ed25519),
-        _ => Err(CryptoError::UnsupportedSignatureScheme),
-    }
-}
+    // TODO: should we return an error on unsupported AES algorithms here?
```

Comment:
I don't think so. My reading is that this is just about mapping the algorithm enums, not about support.